File: JamFix.Services/Database/Zahtjev.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JamFix.Services.Database
{
    public class Zahtjev
    {
        public int ZahtjevId { get; set; }
        public string ImePrezime { get; set; }
        public string Adresa { get; set; }
        public string BrojTelefona { get; set; }
        public string Opis{ get; set; }
        public DateTime DatumVrijeme { get; set; }
        public bool HitnaIntervencija { get; set; }
        public int StatusZahtjevaId { get; set; }
        public virtual StatusZahtjeva StatusZahtjeva{ get; set; }

    }
}

File: RS2/UI/jamfix_admin/lib/models/zahtjev.dart
import 'package:jamfix_admin/models/statusZahtjeva.dart';
import 'package:json_annotation/json_annotation.dart';

part 'zahtjev.g.dart';

@JsonSerializable()
class Zahtjev {
  int? zahtjevId;
  String? imePrezime;
  String? opis;
  String? adresa;
  DateTime? datumVrijeme;
  String? brojTelefona;
  bool? hitnaIntervencija;
  StatusZahtjeva? statusZahtjeva;
  int? statusZahtjevaId;

  Zahtjev(
      {this.imePrezime,
      this.zahtjevId,
      this.adresa,
      this.brojTelefona,
      this.opis,
      this.datumVrijeme,
      this.hitnaIntervencija,
      this.statusZahtjeva,
      this.statusZahtjevaId});

  factory Zahtjev.fromJson(Map<String, dynamic> json) =>
      _$ZahtjevFromJson(json);

  Map<String, dynamic> toJson() => _$ZahtjevToJson(this);
}
